packages/ra-ui-materialui/src/button/SaveButton.spec.tsx

```diff
@@ -400,4 +405,62 @@ describe('<SaveButton />', () => {
             )
         );
     });
+
+    it('should not be enabled if no inputs have changed', async () => {
+        render(
+            <AdminContext dataProvider={testDataProvider()}>
+                <SimpleForm
+                    resource="myresource"
+                    onSubmit={jest.fn}
+                    defaultValues={{
+                        test: 'test',
+                        arr: [
+                            {
+                                id: 123,
+                                foo: 'bar',
+                                nested: [
+                                    {
+                                        deep: '1',
+                                    },
+                                ],
+                            },
+                            {
+                                id: 456,
+                                foo: 'baz',
+                                nested: [
+                                    {
+                                        deep: '2',
+                                    },
+                                ],
+                            },
+                        ],
+                    }}
+                >
+                    <TextInput source="test" />
+                    <ArrayInput resource="foo" source="arr">
+                        <SimpleFormIterator>
+                            <NumberInput source="id" />
+                            <ArrayInput resource="bar" source="arr.nested">
```

Comment:
Why do you need two tested arrayinputs to test that case?

Reply:
You're right! I can simplify it